File: 240430/작은 구슬의 이동/small-marble-movement.py
dx,dy=[0,1,-1,0],[1,0,0,-1]

mapper={
    'R':0,
    'D':1,
    'U':2,
    'L':3
}

def in_range(x,y):
    return 0<x<=n and 0<y<=n

n,t=map(int,input().split())
x,y,d=input().split()
x,y=int(x),int(y)
cur_d=mapper[d]

for _ in range(t):
    nx,ny=x+dx[cur_d],y+dy[cur_d]
    if not in_range(nx,ny):
        cur_d=3-cur_d
        continue
    x,y=nx,ny 

print(x,y)

File: 251210/작은 구슬의 이동/small-marble-movement.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n,t] = input[0].split(' ').map(Number);
const [r,c,d] = input[1].split(' ');

let x = Number(r), y = Number(c);

const dx = [0, 1, -1, 0], dy = [1, 0, 0, -1];

function isValid(x,y){
    return 1<=x && x<=n && 1<=y && y<=n;
}

const mapper = {
    'U':2,
    'D':1,
    'R':0,
    'L':3
};

let dir = mapper[d];

for(let i=0; i<t; i++){
    let nx = x+dx[dir], ny = y+dy[dir];
    if(isValid(nx,ny)){
        x=nx, y=ny;
    }else{
        dir = 3-dir;
    }

}

console.log(x,y)